cdap-docs/reference-manual/source/http-restful-api/metadata.rst

```diff
@@ -1207,6 +1208,336 @@ Rolling up the above using ``rollup=workflow`` would group the programs together
 
 .. highlight:: console
 
+.. _http-restful-api-metadata-fieldlevellineage:
+
+Field Level Lineage
+===================
+
+Fields associated with the Dataset
+----------------------------------
+
+Get the fields which were associated with the dataset in a given time range::
+
+  GET /namespaces/{namespace-id}/datasets/{dataset-id}/lineage/fields?start=<start-ts>&end=<end-ts>[&prefix=<prefix>]
+
+where:
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``namespace-id``
+     - Namespace ID
+   * - ``dataset-id``
+     - Name of the ``dataset``
+   * - ``start-ts``
+     - Starting time-stamp (inclusive), in seconds. Supports ``now``, ``now-1h``, etc. syntax.
+   * - ``end-ts``
+     - Ending time-stamp (exclusive), in seconds. Supports ``now``, ``now-1h``, etc. syntax.
+   * - ``prefix``
+     - Optional ``prefix``, when provided only fields that have given prefix will be returned
+
+|
+
+Following is sample response::
+
+  ["firstName","lastName","item","id","customer_id"]
+
+.. highlight:: console
+
+.. rubric:: HTTP Responses
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Status Codes
+     - Description
+   * - ``200 OK``
+     - Fields of dataset are returned as a list of strings in the body of the response
+   * - ``400 BAD REQUEST``
+     - Failure to parse the time range provided
+
+Field Lineage Summary
+---------------------
+
+Get the field lineage summary for a specified field of a dataset. Field lineage summary consists of set of
+dataset and fields belonging to them which participated in the computation of a given field of a dataset::
+
+  GET /namespaces/{namespace-id}/datasets/{dataset-id}/lineage/fields/{field-name}?start=<start-ts>&end=<end-ts>&direction=incoming
+
+where:
+
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``namespace-id``
+     - Namespace ID
+   * - ``dataset-id``
+     - Name of the ``dataset``
+   * - ``field-name``
+     - Name of the ``field``
+   * - ``start-ts``
+     - Starting time-stamp (inclusive), in seconds. Supports ``now``, ``now-1h``, etc. syntax.
+   * - ``end-ts``
+     - Ending time-stamp (exclusive), in seconds. Supports ``now``, ``now-1h``, etc. syntax.
+   * - ``direction``
+     - ``incoming``, to return the set of dataset and fields which participated in the computation of the given field
+
+|
+
+Returned response consists of direction in which the summary is requested (currently only ``incoming`` direction is
```

Comment:
The returned response consists of the direction in which the summary is requested and the datasets and fields that were responsible for the computation of a specified field. Currently, the only supported direction is `incoming`.

Reply:
done.